crates/ruff_python_ast/src/nodes.rs

```diff
@@ -1891,7 +1891,7 @@ impl From<FStringFlags> for AnyStringFlags {
     }
 }
 
-#[derive(Clone, Debug, PartialEq, is_macro::Is)]
```

Comment:
I don't think deriving `PartialOrd` on `Number` is correct because `Float(10)` is now larger than `Int(10000)`

Reply:
The implementation derived by `PartialEq`:

* If both left and right are of the same variant: compare the inner values
* Otherwise: compare the variant index only

Reply:
Right I just had another look at the docs thanks!

I'll see if I can manually define `PartialOrd` on `Number`, unless you had other suggestions?

Reply:
The problem is that `Number` can be a `str`, which makes comparisons extremely awkward. I guess we could return `None` in that case (unless they're `Equal`). However, I'm not convinced that we want to make comparing `Number` easy. I think it's good that callers have to think through those cases.